site/zenodo_rdm/legacy/serializers/schemas.py

```diff
@@ -63,6 +171,20 @@ def dump_resource_type(self, result, original, **kwargs):
                 result[f"{upload_type}_type"] = resource_type_id.split("-")[-1]
         return result
 
+    @post_dump(pass_original=True)
+    def dump_contributor(self, result, original, **kwargs):
+        """Dumps contributor."""
+        contributors = result.get("contributors", [])
+        new_contributors = []
+        for c in contributors:
+            contributor_type = c.get("type")
+            # Supervisors are serialized in "thesis", thus are removed from contributors list
```

Comment:
A bit of a bamboozle, but in legacy you can also have normal contributors that are supervisors :)

- [`contributors.type:Supervisor`](https://zenodo.org/search?page=1&size=20&q=contributors.type:Supervisor)
- [`contributors.type:Supervisor AND _exists_:thesis`](https://zenodo.org/search?page=1&size=20&q=contributors.type:Supervisor%20AND%20_exists_:thesis): this is even worse, basically there's duplication between the fields, and sometimes even mismatches...

I can't think of an elegant solution for handling this and preserve the user input... Maybe the solution that is "consistent" with the reality is to serialize everything as contributors (and never return `thesis.supervisors` basically).